libs/graphslam/include/mrpt/graphslam/CGraphSlamEngine_impl.h

```diff
@@ -2575,16 +2532,14 @@ template <class GRAPH_T>
 void CGraphSlamEngine<GRAPH_T>::initSlamMetricVisualization()
 {
 	ASSERTDEB_(m_enable_visuals);
-	using namespace std;
-	using namespace mrpt::gui;
 
 	MRPT_START;
 	MRPT_LOG_DEBUG_STREAM("In initializeSlamMetricVisualization...");
 	ASSERTDEB_(m_visualize_SLAM_metric);
 
 	// initialize the m_win_plot on the stack
-	m_win_plot = new CDisplayWindowPlots(
-		"Evolution of SLAM metric - Deformation Energy (1:1000)", 400, 300);
+	m_win_plot.reset(new mrpt::gui::CDisplayWindowPlots(
```

Comment:
I'd prefer `std::make_shared<mrpt::gui::CDisplayWindowPlots>("Evolution of SLAM metric - Deformation Energy (1:1000)", 400, 300))`;

Reply:
I of course meant std::make_unique